bmSettings.lua

```diff
@@ -34,6 +38,369 @@ BMSettings.Constants.UpdateRates = {
+
+    self:saveToDB(db, "INSERT INTO characters (character, character_locked, character_hide_title) VALUES (?, ?, ?)",
+        charName, charData.Locked and 1 or 0, charData.HideTitleBar and 1 or 0)
+
+    if charData.Windows then
+        for windowID, windowData in ipairs(charData.Windows or {}) do
+            windowData.Pos = windowData.Pos or { x = 0, y = 0, } -- Default position
+            for setIndex, setName in ipairs(windowData.Sets or {}) do
+                self:saveToDB(db, [[
+                    INSERT OR REPLACE INTO windows (
+                        server, character, window_id, window_fps, window_button_size, window_advtooltip,
+                        window_compact, window_hide_title, window_width, window_height, window_x, window_y,
+                        window_visible, window_font_size, window_locked, window_theme, window_set_id, window_set_name
+                    ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]],
+                    mq.TLO.EverQuest.Server(), charName, windowID, windowData.FPS or 0, windowData.ButtonSize or 0, windowData.AdvTooltips and 1 or 0,
+                    windowData.CompactMode and 1 or 0, windowData.HideTitleBar and 1 or 0, windowData.Width or 0, windowData.Height or 0,
+                    windowData.Pos.x or 0, windowData.Pos.y or 0, windowData.Visible and 1 or 0, windowData.Font or 0,
+                    windowData.Locked and 1 or 0, windowData.Theme or "", setIndex, setName
+                )
+            end
+        end
+    end
+    db:close()
+end
+
+function BMSettings:deleteButtonFromDB(id)
+    local db = self:InitializeDB()
+    self:saveToDB(db, "DELETE FROM buttons WHERE button_number = ?", id)
+    db:close()
+end
+
+function BMSettings:deleteSetFromDB(setName)
+    local db = self:InitializeDB()
+    self:saveToDB(db, "DELETE FROM sets WHERE set_name = ?", setName)
+    db:close()
+end
+
+function BMSettings:deleteButtonFromSetDB(setName, buttonNumber)
+    local db = self:InitializeDB()
+    self:saveToDB(db, "DELETE FROM sets WHERE set_name = ? AND button_number = ?", setName, buttonNumber)
+    db:close()
+end
+
+function BMSettings:deleteSetFromCharacterDB(charName, SetName)
+    local db = self:InitializeDB()
+    self:saveToDB(db, "DELETE FROM windows WHERE character = ? AND window_set_name = ?", charName, SetName)
+    db:close()
+end
+
+-- Main Function to Convert Config to DB
+function BMSettings:convertConfigToDB(table_name)
+    if table_name == nil then table_name = "all" end
+    local db = self:InitializeDB()
+    local config, err = loadfile(configFile)
+    if err or not config then
+        print("Error loading config file:", err)
+        return
+    end
+
+    local settings = config()
+
+    -- Save Global Settings
+    if table_name == "all" or table_name == "global" then
+        self:saveToDB(db, "INSERT OR REPLACE INTO settings (server, character, settings_version, settings_last_backup) VALUES (?, ?, ?, ?)",

```

Comment:
this likely will result in duplicates for characters as it does not specify the PK